Backend/src/test/java/com/example/Backend/controller/UserControllerTest.java

```diff
@@ -0,0 +1,115 @@
+    @Mock
+    LoginServiceImpl loginService;
+    @Mock
+    ContactServiceImpl contactService;
+    @InjectMocks
+    UserController userController;
+
+    @Test
+    void itShould_DisplayUser() {
+        UserLogin userLogin = new UserLogin();
+        userLogin.setUsername("rs3523");
+        userLogin.setPassword("Rohit@123");
+        when(loginService.getUserDetails(userLogin.getUsername())).thenReturn(userLogin);
+        ResponseEntity<?> responseEntity = userController.displayUser("rs3523");
+        assertEquals(HttpStatus.OK, responseEntity.getStatusCode());
+        assertEquals(userLogin, responseEntity.getBody());
+    }
+
+    @Test
+    void itShould_DisplayUserContact() {
+        UserContact userContact1 = new UserContact("rs3523","9568000766");
+        UserContact userContact2 = new UserContact("rs3523","9999977777");
+        List<UserContact> expectedUserContactsList = new ArrayList<>();
+        expectedUserContactsList.add(userContact1);
+        expectedUserContactsList.add(userContact2);
+        when(contactService.getContactDetails("rs3523")).thenReturn(expectedUserContactsList);
+        ResponseEntity<?> result = userController.displayUserContact("rs3523");
+        log.info("Result: "+ result);
+        assertEquals(HttpStatus.OK, result.getStatusCode());
+        assertEquals(result,new ResponseEntity<>(expectedUserContactsList,HttpStatus.OK));
+    }
+
+    @Test
+    void itShould_LoginUser() {
+        UserLogin userLogin = new UserLogin();
+        userLogin.setUsername("rs3523");
+        userLogin.setPassword("Rohit@123");
+        ResponseEntity<?> result = userController.loginUser(userLogin);
+        log.info("Logged in : "+ result);
+        assertEquals(HttpStatus.ACCEPTED, result.getStatusCode());
+    }
+
+    @Test
+    void itShould_RegisterUser() {
+        UserRegister userRegister = new UserRegister();
+        userRegister.setUsername("rs3523");
+        userRegister.setPassword("Rohit@123");
+        userRegister.setEmail("rohitsaini@gmail.com");
+        ResponseEntity<?> result = userController.registerUser(userRegister);
+        log.info("Register user: "+ result);
+        assertEquals(HttpStatus.CREATED, result.getStatusCode());
+    }
+
+    @Test
+    void itShould_AddContactDetails() {
+        UserContact userContact = new UserContact();
+        userContact.setUsername("rs3523");
+        userContact.setPhoneNumber("9988007722");
+        ResponseEntity<?> result = userController.addContactDetails("rs3523", userContact);
+        assertEquals(HttpStatus.CREATED, result.getStatusCode());
+    }
+
+    @Test
+    void itShouldThrow_InvalidInputExceptionFor_AddContactDetails() {
+        UserContact userContact = new UserContact();
+        userContact.setUsername("invalidUsername");
+        userContact.setPhoneNumber("0000000000");
+        when(contactService.addContactDetails("invalidUsernam",userContact)).thenThrow(new InvalidInputException("Invalid Request"));
```

Comment:
For exception you should have a Controller advice test class. You assert the response for the exception that is through and the error response you send back